### DISCUSSION TITLE
ESTKALMAN: State out of bounds, resetting

### DISCUSSION THREAD

**Discussion by AlEXANDER9901** [bitcraze/discussions#562] [Q&A]
hi!

We are going to fly a drone using loco positioning. I am following getting start, https://www.bitcraze.io/documentation/tutorials/getting-started-with-loco-positioning-system/#click-the-loco-positioning-tab

There seems to be a problem with the link above.
Although cfclient and crazyflie are connected, all anchor positions are shown as the origin.
The position of crazyflie is also not fixed and moves.
I checked the console tab and got "ESTKALMAN: State out of bounds, resetting ".

Additionally, we are working on the Bitcraze VM.

Can you help me with what the problem is?

![사본 -026](https://user-images.githubusercontent.com/125459416/219014321-33e81eb8-5ee3-4edc-b0fa-5164f3128a2d.png)
![사본 -025](https://user-images.githubusercontent.com/125459416/219014503-e8fd0c94-2001-4637-ba53-b009e466c4d0.png)

**Comment by hmllr** [comment]
Hi :)
What are you exactly trying to achieve? Localizing the Crazyflie? And how is your system set up (like how many anchors in what positions?)?

You seem to be using only one anchor from the printscreen - or at least only one is found. This will not work to localize the Crazyfle - you always only get one distance measurement per anchor, so you need at the very least 4 (well-placed) anchors to get a stable position.

Hope this helps,
Hanna

**Reply by AlEXANDER9901** [reply]
thanks for reply. 

https://www.bitcraze.io/2022/12/the-big-loco-show/#comments
we're trying to follow this.

Anchors are well set up from 0 to 7 according to the rules.
Also, the Command Based Flight Control window does not open even when connected to the drone. The xyz axis in state estimate is constantly changing. And, on the anchor, it is displayed like a video.
We have completed all updates and installations in tutorial.

Is there a communication problem between the loco positoning deck and the anchor?
Drone position is not fixed in cfclient. Could there be a loco positioing deck problem?

![00](https://user-images.githubusercontent.com/125459416/219033388-c7f1d6a8-cf9f-44f5-83cd-15e971ee4f91.png)

https://user-images.githubusercontent.com/125459416/219031635-96df2cd6-b4fd-442f-9473-64ac730a62eb.mp4

**Reply by hmllr** [reply]
Did you program different IDs in all your anchors, as written in the tutorial? According to your screenshot it only finds an anchor 0 (the only green one), so localization is not successful as this doesn't work with only one anchor. 
This problem has to be tackled first, before all anchors are green and configured with the right position flying will only end up in a crash anyway.
However, for helping with the greyed out flight control it would be useful to send a screenshot of the "console" tab as well.
Hope this helps!

**Reply by AlEXANDER9901** [reply]
thanks I'll try~!

**Comment by krichardsson** [comment]
The commander buttons are activated if you have a positioning deck mounted (which you have) but there must also NOT be a gamepad connected as this can give conflicting commands to the Crazyflie. If you hover the mouse over the buttons, you should get a tool tip telling you what you need to do to activate the buttons.

**Comment by krichardsson** [comment]
(Copied from #564)

thanks for reply.

In fact, we have installed 8 anchors in the room, and we want to fly the drone with Loco positioning, but as shown in the picture, the anchors appear green, but the location is not accurate.
So I wondered if there was a problem with transmitting and receiving data in the anchor.
but thanks to that, it was solved.

Again, we look at the console window to see if there is a problem with the LPS deck. I'd like to find out if there's a procedure we're missing, or if a deck setup is needed.
We think there is a problem with the loco positioning deck mounting, but we are curious because we are not sure.
![사본 -026](https://user-images.githubusercontent.com/125459416/219430428-4c5fad55-0c72-4d04-ba7f-6c0f3d3d20e0.png)
![dhkdhfkdhfkdhfdkhfd](https://user-images.githubusercontent.com/125459416/219430460-cbddfaf2-e74a-40a1-9bde-d0ec28db08b8.png)

As an additional note, in the procedure we followed, there was no special setting for the positioning deck.
https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/howto/
Do I need this course?

I would like it if you could help me.

**Reply by krichardsson** [reply]
It looks like your anchors are using the correct IDs, except anchor 1 that is not seen. As the boxes are green we know the deck is working and communicating with the anchors (that are also fine).

In the 3D graph I can not see dots representing anchors which indicates that you have not [configured the position of the anchors properly](https://www.bitcraze.io/documentation/tutorials/getting-started-with-loco-positioning-system/#enter-anchor-positions). When you are done you should get a 3D view with the anchors as in the tutorial

You get the warning about the estimator resetting as a result of the anchor positions, it simply means that the estimator can not make sense of the data it gets. 

[There is also an overview of the Loco system](https://www.bitcraze.io/documentation/system/positioning/loco-positioning-system/) that is a good starting point to understand how it works

**Reply by MINJEEgi** [reply]
hi :)  is there a way to resolve the warning about estimator resetting error? we have same consol error

**Reply by krichardsson** [reply]
The "State out of bounds, resetting" message simply means that the estimator can not make sens of the data it gets. When input data is "bad" the estimated state (usually) diverges quickly and takes off towards infinity. The estimator is reset when the state is unreasonable, either the position is far away (+-100 meters) or the velocity is very high (10 m/s). There can be many reasons this happens, depending on positioning system, setup, environment, configuration and so on, and there is not one simple answer to why it happens. 

@MINJEEgi If you get resets and can not find the solution in this thread or other existing threads, please start a new discussion describing your setup and problem.
